src/main/java/com/google/api/gax/bundling/AccumulatingBundleReceiver.java

```diff
@@ -48,7 +48,9 @@ public void validateItem(T message) {
 
   @Override
   public void processBundle(List<T> bundle) {
-    bundles.add(bundle);
+    List<T> bundleCopy = new ArrayList<>();
+    bundleCopy.addAll(bundle);
+    bundles.add(bundleCopy);
```

Comment:
I think we could combine 51 and 52 in 1 line: List<T> bundleCopy = new ArrayLiast<>(bundle)